include/envoy/grpc/status.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+namespace Envoy {
+namespace Grpc {
+
+enum GrpcStatus {
+  Ok = 0,
+  Canceled = 1,
+  Unknown = 2,
+  InvalidArgument = 3,
+  DeadlineExceeded = 4,
+  NotFound = 5,
+  AlreadyExists = 6,
+  PermissionDenied = 7,
+  Unauthenticated = 16,
+  ResourceExhausted = 8,
+  FailedPrecondition = 9,
+  Aborted = 10,
+  OutOfRange = 11,
+  Unimplemented = 12,
+  Internal = 13,
+  Unavailable = 14,
+  DataLoss = 15,
+  InvalidCode = 16,
```

Comment:
Both Unauthenticated and InvalidCode get same value (16)?
I don't think you need InvalidCode, as if grpc expand the enum in the future, it introduces conflict for sure.
Also, please copy all the comments. And add one additional comment to indicate that this enum should be kept sync with grpc.